src/foxsisim/module.py

```diff
@@ -124,8 +140,19 @@ def passRays(self, rays, robust=False):
                 regions[i].extend(self.shells[i + 1].getSurfaces())
 
         for ray in rays:
-            # move ray to the front of the optics
-            ray.moveToZ(self.coreFaces[0].center[2])
+            # Does the ray hit the front sieve?
+            for annulus in self.sieveFront:
```

Comment:
maybe create a function to check if a ray enters the optic which checks whether it hits the blocker or side of a shell and same for a function for it to exit? There may be other objects that block the path in and out like the spider spokes.